CMakeLists.txt

```diff
@@ -196,7 +196,10 @@ if (NOT ANDROID)
         set(JAVA_AWT_LIBRARY NotNeeded)
         set(JAVA_AWT_INCLUDE_PATH NotNeeded)
     endif()
+    file(TO_CMAKE_PATH "$ENV{CMAKE_JAVA_HOME}" CMAKE_JAVA_HOME)
+    set(JAVA_HOME "${CMAKE_JAVA_HOME}") # required because gradle sets JAVA_HOME to the JRE path
```

Comment:
I think we use maven to build an release, why gradle will be invoked here?

Reply:
Looks like the CI calls it from here? https://github.com/awslabs/aws-crt-java/blob/main/builder.json#L39 . Definitely it is causing [some checks to fail (e.g. linux-compat al2-x64)](https://github.com/awslabs/aws-crt-java/runs/4657370335?check_suite_focus=true)

I'll move this to a distinct PR and address the comments there, thanks!

Reply:
Addressed in PR #423